packages/ui/src/council/modals/AnnounceCandidacy/components/TitleAndBulletPointsStep.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import * as Yup from 'yup'
+
+import { InputComponent, InputText } from '@/common/components/forms'
+import { Row } from '@/common/components/Modal'
+import { RowGapBlock } from '@/common/components/page/PageContent'
+import { useForm } from '@/common/hooks/useForm'
+
+interface TitleAndBulletPointsStepProps {
+  title?: string
+  bulletPoints: string[]
+  setTitle: (title?: string) => void
+  setBulletPoints: (bulletPoints: string[]) => void
+}
+
+interface FormFields {
+  title?: string
+  bulletPoint1?: string
+  bulletPoint2?: string
+  bulletPoint3?: string
+}
+
+type StringType = 'title' | 'bulletPoint'
+
+const typeLength: { [key in StringType]: number } = { title: 60, bulletPoint: 120 }
+
+const FormSchema = Yup.object().shape({})
+
+export const TitleAndBulletPointsStep = ({
+  title,
+  bulletPoints,
+  setTitle,
+  setBulletPoints,
+}: TitleAndBulletPointsStepProps) => {
+  const formInitializer: FormFields = {
+    title,
+    bulletPoint1: bulletPoints[0],
+    bulletPoint2: bulletPoints[1],
+    bulletPoint3: bulletPoints[2],
+  }
+  const { fields, changeField } = useForm<FormFields>(formInitializer, FormSchema)
+
+  const setValue = (field: keyof FormFields, value: any) => {
+    switch (field) {
+      case 'title':
+        setTitle(!value || isValidString(value, 'title') ? value : undefined)
+        break
+      case 'bulletPoint1':
+      case 'bulletPoint2':
+      case 'bulletPoint3': {
+        const points = [fields.bulletPoint1, fields.bulletPoint2, fields.bulletPoint3].map((point, index) =>
+          index + 1 === Number(field.replace('bulletPoint', '')) ? value : point
+        )
+        const values = points.filter(
+          (bulletPoint) => bulletPoint && !!bulletPoint?.trim() && isValidString(bulletPoint, 'bulletPoint')
+        ) as string[]
```

Comment:
This is done to avoid empty bullet points

Reply:
I think that this should be possible to achieve with `Yup` or by using static validation. I'll see what we can do here with Yup.

I'm worried that this part looks a bit hackish. And some simpler solution should be possible.